### PR TITLE
- Reldens - v4.0.0-beta.26 - Show terms and conditions

### PR DESCRIPTION
**Pull request by damian-pastorini**

### CODE REVIEW

File: theme/default/index.js [hunk 1]
```diff
@@ -115,6 +119,32 @@ window.addEventListener('DOMContentLoaded', () => {
         });
     }
 
+    let termsConditionsContainer = dom.getElement('#terms-conditions');
+    register.querySelector('#term_conditions').addEventListener('click', (ev) => {
+        ev.preventDefault();
+        termsConditionsContainer.classList.remove('hidden');
+        reldens.gameDom.getJSON(reldens.appServerUrl + '/terms-and-conditions', (err, response) => {
+            console.log(err);
+            console.log(response);
```
Comment (damian-pastorini) [inline theme/default/index.js:128]: remove the console.log(s)

File: theme/default/index.js [hunk 2]
```diff
@@ -115,6 +119,32 @@ window.addEventListener('DOMContentLoaded', () => {
         });
     }
 
+    let termsConditionsContainer = dom.getElement('#terms-conditions');
+    register.querySelector('#term_conditions').addEventListener('click', (ev) => {
+        ev.preventDefault();
+        termsConditionsContainer.classList.remove('hidden');
+        reldens.gameDom.getJSON(reldens.appServerUrl + '/terms-and-conditions', (err, response) => {
```
[the remark this answers is not in the document]
Comment (damian-pastorini) [reply theme/default/index.js:126]: also, if there's none content the link should not be displayed

File: theme/default/index.js [hunk 3]
```diff
@@ -115,6 +119,32 @@ window.addEventListener('DOMContentLoaded', () => {
...
+            console.log(err);
+            console.log(response);
+            termsConditionsContainer.querySelector('.loading-container').style.display = 'none';
+            termsConditionsContainer.querySelector('.tacbox').style.display = 'block';
```
Comment (damian-pastorini) [inline theme/default/index.js:130]: not sure if node.querySelector was used in the rest of the file but try to use gameDom.getElement instead

File: lib/game/server/login-manager.js [hunk 4]
```diff
@@ -288,6 +297,15 @@ class LoginManager
         });
     }
 
+    fetchTermsAndConditions()
+    {
+        let heading = this.config.get('client/login/terms_and_conditions/heading', false);
+        let body = this.config.get('client/login/terms_and_conditions/body', false);
```
[the remark this answers is not in the document]
Comment (luciovicentini) [reply lib/game/server/login-manager.js:303]: oohhh.... nice trick! didn't know that!

File: migrations/production/beta.26-sql-update.sql [hunk 5]
```diff
@@ -203,6 +203,28 @@ INSERT INTO `objects_items_rewards_animations` VALUES
...
     (2, 1, 'spritesheet', 'branch-sprite', 'branch-sprite', '{"start":0,"end":2,"repeat":-1,"frameWidth":32, "frameHeight":32,"depthByPlayer":"above"}');
 
+
+INSERT INTO `config` VALUES (NULL, 'client', 'login/terms_and_conditions/body', 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent vitae laoreet est, eget mattis justo. Donec urna sapien, tincidunt venenatis accumsan eu, molestie at ante. Suspendisse sed semper lacus. Mauris porttitor urna id odio bibendum, non ornare justo finibus. Donec vestibulum condimentum ante, non malesuada nulla porta eu. Proin scelerisque nibh at dolor mattis, sed laoreet nibh gravida. Interdum et malesuada fames ac ante ipsum primis in faucibus. In hac habitasse platea dictumst. Nullam euismod lorem vel imperdiet porttitor. Sed sapien massa, dapibus vitae ante ut, elementum scelerisque diam. Quisque a tempor dui. Suspendisse semper, nibh in faucibus tincidunt, orci est sollicitudin magna, eu ullamcorper dui lacus non ipsum. Praesent vitae orci a eros maximus condimentum ac a quam. In sodales velit lorem, id finibus lacus tempor quis.
+
```
[the remark this answers is not in the document]
Comment (luciovicentini) [reply migrations/production/beta.26-sql-update.sql:208]: ok!

File: theme/default/index.js (hunk 1 above)
Comment (luciovicentini) [reply theme/default/index.js:128]: my bad!

File: theme/default/index.js (hunk 3 above)
Comment (luciovicentini) [reply theme/default/index.js:130]: yep... it is used all over the file. I'm going to change it on my code.

File: theme/default/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (luciovicentini) [reply theme/default/index.js:126]: Wait... If we going to make the call when the page is loaded, isn't a way to send data from the server when the page is served?

**Pull request merged**